lib/srv/desktop/windows_server.go

```diff
@@ -428,7 +435,7 @@ func (s *WindowsService) newStreamer(ctx context.Context, recConfig types.Sessio
 		return nil, trace.Wrap(err)
 	}
 
-	return libevents.NewTeeStreamer(fileStreamer, s.cfg.Emitter), nil
```

Comment:
I'm confused -- does this mean we never needed to send recording events to the `s.cfg.Emitter` in `async` mode? We can just write them to disk, and they will get uploaded later by some other process?

Related comment regarding the overall changes: my understanding is that session recording events and audit events are going to the same place. If that's the case, then it seems that we shouldn't need a separate audit event emitter and recording event emitter. (I believe this is what we had before these changes, although our naming/lack-of-documentation made it a bit difficult to track).

Perhaps related observation: The `s.cfg.AuthClient` that we return in `sync` mode is the same `*auth.Client` as the `s.cfg.Emitter` that we were passing to the `TeeStreamer` previously, see [here](https://github.com/gravitational/teleport/blob/8571856/lib/service/desktop.go#L210-L215).

Reply:
> my understanding is that session recording events and audit events are going to the same place

This is an easy mistake to make due to the way we've set things up, but your understanding is incorrect. Audit events are written to Teleport's audit backend (typically DynamoDB) via `cfg.Emitter`. Session recordings are written to the sessions backend (typically S3) via the `StreamWriter`.

Despite the different destinations, events are recorded using the `EmitAuditEvent` function in both cases (I'm really not a fan of this approach as it causes confusion)

The `TeeStreamer` is attempting to be clever, and I don't think clever is good here. Essentially it exposes a single `EmitAuditEvent` entry point, and it sends every event it receives to the recording (AKA stream), and a subset of the events (based on the event type) to Teleport's audit log.

What we're doing in this first commit by removing the `TeeStreamer` is moving the logic for which desktop events belong in session recordings and which desktop events belong in the audit log out of a shared component and into the `WindowsService` which knows best.

Reply:
I see. So before these changes, when we had

```go
func (s *WindowsService) newStreamer(ctx context.Context, recConfig types.SessionRecordingConfig) (libevents.Streamer, error) {
	if services.IsRecordSync(recConfig.GetMode()) {
		s.cfg.Log.Debugf("using sync streamer (for mode %v)", recConfig.GetMode())
		return s.cfg.AuthClient, nil
	}
	s.cfg.Log.Debugf("using async streamer (for mode %v)", recConfig.GetMode())
	uploadDir := filepath.Join(s.cfg.DataDir, teleport.LogsDir, teleport.ComponentUpload,
		libevents.StreamingSessionsDir, apidefaults.Namespace)
	fileStreamer, err := filesessions.NewStreamer(uploadDir)
	if err != nil {
		return nil, trace.Wrap(err)
	}

	return libevents.NewTeeStreamer(fileStreamer, s.cfg.Emitter), nil
}
```

In that case, in `sync` recording mode, we would just send all audit and recording events directly to the auth server. In `async` mode, we would send [all audit and recording events to the `fileStreamer`](https://github.com/gravitational/teleport/blob/9fa28188c75c44baf9344fed1f1b2738577519f5/lib/events/emitter.go#L498), and [all audit, but not recording, events to the `s.cfg.Emitter` (aka the auth server)](https://github.com/gravitational/teleport/blob/9fa28188c75c44baf9344fed1f1b2738577519f5/lib/events/emitter.go#L503-L508).

With your proposed changes, `sync` mode still works the same (everything is sent to auth), however in `async` mode, _only recording events_ will be sent to the `fileStreamer`, no longer audit events. Audit events are now only sent to the auth server in `async` mode.